libp2p/NodeTable.cpp

```diff
@@ -328,50 +315,46 @@ void NodeTable::noteActiveNode(Public const& _pubk, bi::udp::endpoint const& _en
     if (_pubk == m_node.address() || !NodeIPEndpoint(_endpoint.address(), _endpoint.port(), _endpoint.port()).isAllowed())
         return;
 
-    shared_ptr<NodeEntry> node = nodeEntry(_pubk);
-    if (!!node && !node->pending)
+    shared_ptr<NodeEntry> newNode = nodeEntry(_pubk);
+    if (newNode && !newNode->pending)
     {
         clog(NodeTableConnect) << "Noting active node:" << _pubk << _endpoint.address().to_string() << ":" << _endpoint.port();
-        node->endpoint.address = _endpoint.address();
-        node->endpoint.udpPort = _endpoint.port();
-        
-        shared_ptr<NodeEntry> contested;
+        newNode->endpoint.address = _endpoint.address();
+        newNode->endpoint.udpPort = _endpoint.port();
+
+        shared_ptr<NodeEntry> nodeToEvict;
         {
+            // Find a bucket to put a node to
```

Comment:
See also @fjl's description of what is happening here at https://github.com/ethereum/devp2p/blob/master/discv4.md#node-table